checker/src/call_visitor.rs

```diff
@@ -2108,7 +2119,9 @@ impl<'call, 'block, 'analysis, 'compilation, 'tcx, E>
                 match rustc_gen_args[1].unpack() {
                     GenericArgKind::Type(rustc_type) => tag_rustc_type = rustc_type,
                     _ => {
-                        assume_unreachable!("expected the second generic argument of tag-related function calls to be a type");
+                        assume_unreachable!(
```

Comment:
Assumptions should be motivated in a comment. WHY is it reasonable to expect the second generic argument to be a type? Where is this checked?

Reply:
These assumptions are documented by this function's documentation and also a comment on line 2117. This is ensured by the rust type checker: the second argument of add_tag!/has_tag! macros must be a type.